api/api.go

```diff
@@ -948,3 +948,569 @@ func (f *PublicController) SearchCourse(c *gin.Context) {
+		log.Errorf("Unable to get role_id from Cookie: %s", err.Error())
+		c.IndentedJSON(http.StatusUnauthorized, err.Error())
+		return
+	}
+
+	if !AuthorizeModerator(role_id) {
+		log.Infof("User is not authorized: %s", err.Error())
+		c.Status(http.StatusUnauthorized)
+		return
+	}
+
+	raw, err := c.GetRawData()
+	if err != nil {
+		log.Errorf("Unable to get raw data from request: %s", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	var j map[string]interface{}
+	err = json.Unmarshal(raw, &j)
+	if err != nil {
+		log.Errorf("Unable to unmarshal the json body: %+v", raw)
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+
+	name, ok := j["name"].(string)
+	if !ok {
+		log.Error("unable to convert name to string")
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+	deadline, ok := j["deadline"].(string)
+	if !ok {
+		log.Error("unable to convert deadline to string")
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+	visible_from, ok := j["visible_from"].(string)
+	if !ok {
+		log.Error("unable to convert visible_from to string")
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+	max_filesize, err := strconv.Atoi(j["max_filesize"].(string))
+	//max_filesize, ok := j["max_filesize"].(int)
+	if err != nil {
+		log.Error("unable to convert maxfilesize to int")
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+
+	_, err = course.EditSubmission(f.Database, id, name, deadline, max_filesize, visible_from)
+	if err != nil {
+		log.Errorf("Unable to update submission: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, id)
+}
+
+func (f *PublicController) GetSubmissionFromUser(c *gin.Context) {
+
+	user_id, err := f.GetIdFromCookie(c)
+	if err != nil {
+		log.Errorf("Unable to get id from Cookie: %s", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	// Fetch Data from Database with Backend function
+	users, err := course.GetSubmissionsFromUser(f.Database, user_id)
+	if err != nil {
+		log.Errorf("Unable to get submissions from user: %s", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	// Return Status and Data in JSON-Format
+	c.IndentedJSON(http.StatusOK, users)
+}
+
+func (f *PublicController) CreateSubmissionHasFiles(c *gin.Context) {
+
+	role_id, err := f.GetRoleIdFromCookie(c)
+	if err != nil {
+		log.Errorf("Unable to get role_id from Cookie: %s", err.Error())
+		c.IndentedJSON(http.StatusUnauthorized, err.Error())
+		return
+	}
+
+	if !AuthorizeModerator(role_id) {
+		log.Infof("User is not authorized: %s", err.Error())
+		c.Status(http.StatusUnauthorized)
+		return
+	}
+	submission_id, err := strconv.Atoi(c.Param("id"))
+	if err != nil {
+		log.Errorf("Unable to convert parameter `id` to int: %s", err.Error())
+		c.Status(http.StatusBadRequest)
+		return
+	}
+
+	if c.ContentType() == "text/plain" {
+		type _file struct {
```

Comment:
Wennn du das Struct hier auch benutzt, dann deklarier es global in der Datei und benutze es in den 2(?) Funktionen statt es 2 Mal zu deklarieren